chcemvediet/apps/inforequests/tests/test_models/test_deadline.py

```diff
@@ -0,0 +1,157 @@
+
+    def test_calendar_days_behind_property_and_calendar_days_behind_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-26'))
+        self.assertEqual(deadline.calendar_days_behind, 13)
+        self.assertEqual(deadline.calendar_days_behind_at(naive_date(u'2010-10-26')), 13)
+
+    def test_workdays_behind_property_and_workdays_behind_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-26'))
+        self.assertEqual(deadline.workdays_behind, 9)
+        self.assertEqual(deadline.workdays_behind_at(naive_date(u'2010-10-26')), 9)
+
+    def test_is_deadline_missed_property_and_is_deadline_missed_at_method(self):
+        tests = (                            # _today,                    is_deadline_missed
+                (Deadline.UNITS.CALENDAR_DAYS, naive_date(u'2010-10-13'), False),
+                (Deadline.UNITS.CALENDAR_DAYS, naive_date(u'2010-10-14'), True),
+                (Deadline.UNITS.WORKDAYS,      naive_date(u'2010-10-15'), False),
+                (Deadline.UNITS.WORKDAYS,      naive_date(u'2010-10-16'), True),
+        )
+        for unit, _today, is_deadline_missed in tests:
+            deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=unit, _today=_today)
+            self.assertEqual(deadline.is_deadline_missed, is_deadline_missed)
+            self.assertEqual(deadline.is_deadline_missed_at(_today), is_deadline_missed)
+
+    def test_snooze_date_property(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-14'))
+        self.assertEqual(deadline.snooze_date, naive_date(u'2010-10-14'))
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-12'))
+        self.assertEqual(deadline.snooze_date, naive_date(u'2010-10-13'))
+
+    def test_snooze_date_property_default_value_if_omitted(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=None)
+        self.assertEqual(deadline.snooze_date, naive_date(u'2010-10-13'))
+
+    def test_is_snoozed_property(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-14'))
+        self.assertTrue(deadline.is_snoozed)
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-12'))
+        self.assertFalse(deadline.is_snoozed)
+
+    def test_snooze_in_calendar_days_and_snooze_in_workdays_properties(self):
```

Comment:
@viliambalaz By sa zisliel aj pripad, ked `snooze` je nizsie nez povodny deadline. Ze vtedy bude `snooze_in_workdays` nula a nie nejake zaporne cislo.

Reply:
@martinmacko47 Podla mna sa tento stav tyka `snooze_date` property, ktory je otestovany v `test_snooze_date_property`
```
deadline2 = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-12'))
        self.assertEqual(deadline2.snooze_date, naive_date(u'2010-10-13'))
```
ked snooze_date nebude `2010-10-12` ale nastavi sa podla `deadline_date` na `2010-10-13`.
Teraz som to neupravil, ale mozem to zmenit, ako chces.

Reply:
Okey. Nechaj to teda tak ako to je.